### PR TITLE
Add "Ethereum Utils" extension

### PR DESCRIPTION
**Pull request by Destiner**
## Description

Adds a new extension. This extension is a set of utils to help with Ethereum development. It can also be helpful for other EVM-compatible chains (Polygon, BSC, etc).

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/4247901/149595538-56fe7c15-41d2-4171-85e2-6f217cb08e4f.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/ethereum-utils/package.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  "title": "Ethereum Utils — EVM Development",
+  "description": "Collection of utils for Ethereum and EVM-compatible chain development",
+  "icon": "icon.png",
+  "author": "destiner",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "convert-decimals",
+      "title": "Convert Decimal Units",
+      "description": "Converts decimal unit values",
```
Comment (mattisssa) [inline extensions/ethereum-utils/package.json:13]:
```suggestion
      "description": "Converts decimal unit values",
      "subtitle": "Ethereum Utils"
```

**Review by mattisssa — COMMENTED** [review]
Hey! Thanks for the contribution, everything looks good to me.
I think the extension commands need some subtitles to understand what the command is related to. I'm not sure about the one I suggested, but let's add a subtitle to all the commands.

What do you think?

File: extensions/ethereum-utils/src/decode-abi.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        return;
+      }
+    }
+    const coder = new Coder(abi);
+    if (type === 'constructor') {
+      try {
+        const constructor = coder.decodeConstructor(data);
+        push(<DecodedDataView constr={constructor} />);
+      } catch {
+        showToast(ToastStyle.Failure, 'Convertion failed');
```
Comment (pernielsentikaer) [inline extensions/ethereum-utils/src/decode-abi.tsx:74]:
```suggestion
        showToast(ToastStyle.Failure, 'Conversion failed');
```

File: extensions/ethereum-utils/src/decode-abi.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      }
+      return;
+    }
+    if (type === 'function') {
+      try {
+        const func = coder.decodeFunction(data);
+        push(<DecodedDataView func={func} />);
+      } catch {
+        showToast(ToastStyle.Failure, 'Convertion failed');
```
Comment (pernielsentikaer) [inline extensions/ethereum-utils/src/decode-abi.tsx:83]:
```suggestion
        showToast(ToastStyle.Failure, 'Conversion failed');
```

File: extensions/ethereum-utils/src/decode-abi.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      }
+      return;
+    }
+    if (type === 'event') {
+      try {
+        const allTopics = topics.filter((topic): topic is string => !!topic);
+        const event = coder.decodeEvent(allTopics, data);
+        push(<DecodedDataView event={event} />);
+      } catch {
+        showToast(ToastStyle.Failure, 'Convertion failed');
```
Comment (pernielsentikaer) [inline extensions/ethereum-utils/src/decode-abi.tsx:93]:
```suggestion
        showToast(ToastStyle.Failure, 'Conversion failed');
```

File: extensions/ethereum-utils/src/search-chains.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+              {chain.explorers && chain.explorers[0] && (
+                <OpenInBrowserAction
+                  title="Open Chain Explorer"
+                  url={chain.explorers[0].url}
+                  shortcut={{ key: 'e', modifiers: ['cmd'] }}
+                />
+              )}
+            </ActionPanel>
+          }
+        ></ListItem>
```
Comment (pernielsentikaer) [inline extensions/ethereum-utils/src/search-chains.tsx:64]:
```suggestion
        />
```
Collapsing empty tag

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Destiner 👋 

I'm not using anything related to `Ethereum` but it also looks good to me. Only added a few minor suggestions.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/ethereum-utils/package.json (hunk 1 above)
Comment (Destiner) [reply extensions/ethereum-utils/package.json:14]: @mattisssa thanks for suggestion, I agree. I used "Ethereum (EVM) Utils" to allow folks to use "EVM" keyword as well.

**Comment by Destiner** [comment]
Thanks @pernielsentikaer for feedback, I applied all changes.

**Comment by Destiner** [comment]
Hey @pernielsentikaer, wondering if you could provide any update on this. Thanks!

**Comment by pernielsentikaer** [comment]
Hi @Destiner 👋 

Thanks for following up, the only thing I can think of is that the icon might be hard to see when the system is running Dark appearance

LhzoS0oJV3sXLhHB

It's not required to change, but if you have an idea, feel free 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Destiner** [comment]
Thanks for feedback @pernielsentikaer. I updated the icon to add more spacing and contract. Hopefully it looks better now.

**Review by pernielsentikaer — COMMENTED** [review]
Awesome @Destiner 🔥

**Pull request merged**